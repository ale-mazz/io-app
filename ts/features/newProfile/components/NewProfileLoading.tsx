import { LoadingSpinner } from "@pagopa/io-app-design-system";
import React from "react";
import { StyleSheet, View } from "react-native";

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    paddingTop: 16
  }
});

export const NewProfileLoading = () => (
  <View style={styles.container}>
    <LoadingSpinner size={24} />
  </View>
);

export default NewProfileLoading;
