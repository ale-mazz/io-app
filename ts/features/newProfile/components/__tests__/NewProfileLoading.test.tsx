import React from "react";
import { render } from "@testing-library/react-native";
import NewProfileLoading from "../NewProfileLoading";

describe("NewProfileLoading Component", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<NewProfileLoading />);
    expect(toJSON()).toMatchSnapshot();
  });
});
