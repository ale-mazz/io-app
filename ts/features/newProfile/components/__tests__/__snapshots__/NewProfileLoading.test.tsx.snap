// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProfileLoading Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    accessibilityRole="progressbar"
    accessible={true}
    importantForAccessibility="no-hide-descendants"
    style={
      {
        "height": 24,
        "width": 24,
      }
    }
    testID="LoadingSpinnerTestID"
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "rotateZ": "0deg",
            },
          ],
        }
      }
      testID="LoadingSpinnerAnimatedTestID"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                [
                  0,
                  736995,
                  1,
                  -16040221,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="spinner-secondHalf"
              x1="0%"
              x2="100%"
              y1="0%"
              y2="0%"
            />
            <RNSVGLinearGradient
              gradient={
                [
                  0,
                  -16040221,
                  1,
                  -16040221,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="spinner-firstHalf"
              x1="0%"
              x2="100%"
              y1="0%"
              y2="0%"
            />
          </RNSVGDefs>
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "strokeWidth",
              ]
            }
            strokeWidth={3}
          >
            <RNSVGPath
              d="M 1.5 12 A 10.5 10.5 0 0 1 22.5 12"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "brushRef": "spinner-secondHalf",
                  "type": 1,
                }
              }
            />
            <RNSVGPath
              d="M 22.5 12 A 10.5 10.5 0 0 1 1.5 12"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "brushRef": "spinner-firstHalf",
                  "type": 1,
                }
              }
            />
            <RNSVGPath
              d="M 1.5 12 A 10.5 10.5 0 0 1 1.5 11.25"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeLinecap",
                ]
              }
              stroke={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              strokeLinecap={1}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
